galaxy-deploy.yml

```diff
@@ -0,0 +1,50 @@
+---
+# Automated release playbook for Ansible Collections.
+#
+# Originally based on Ericsysmin's 2020 blog post. Meant to be used in a GitHub
+# Actions CI environment.
+#
+# Requires a ANSIBLE_GALAXY_TOKEN secret to be configured on the GitHub repo.
+#
+# Usage:
+#   ansible-playbook -i 'localhost,' galaxy-deploy.yml \
+#   -e "github_tag=${{ github.ref }}"
+
+- hosts: localhost
+  connection: local
+  gather_facts: false
+
+  vars:
+    # Requires github_tag to be set when calling playbook.
+    release_tag: "{{ github_tag.split('/')[-1] }}"
+
+  pre_tasks:
+    - name: Ensure ANSIBLE_GALAXY_TOKEN is set.
+      fail:
+        msg: A valid ANSIBLE_GALAXY_TOKEN must be set.
+      when: "lookup('env','ANSIBLE_GALAXY_TOKEN') | length == 0"
+
+    - name: Ensure the ~/.ansible directory exists.
+      file:
+        path: ~/.ansible
+        state: directory
+
+    - name: Write the Galaxy token to ~/.ansible/galaxy_token
+      copy:
+        content: |
+          token: {{ lookup('env','ANSIBLE_GALAXY_TOKEN') }}
+        dest: ~/.ansible/galaxy_token
+
+  tasks:
+    - name: Ensure the galaxy.yml tag is up to date.
+      lineinfile:
+        path: galaxy.yml
+        regexp: "^version:"
+        line: 'version: "{{ release_tag }}"'
```

Comment:
I don't recommend that.
We could put an assert instead that check the coherence between `current-version.txt`, the detected tag by `release_tag` and the version tag in the galaxy.yml file.
maybe we can drop one of the 3

Reply:
Ok, what I did is I changed now the command to check changed version, since `galaxy.yml` with version is mandatory we don't need `current-version.txt` anymore.

Reply:
Hm. 
So at every merge commit in main, the playbook change version value to main in galaxy.yml. then the build task will generate the archive with main as suffix. This might work  for the build task but for the publish will very probably fail as the version will not be in semver format.
I'll try locally

Reply:
Not sure I understood what you mean. You need to intentionally change the version through your PR, so it will not always be the case of every merge commit, no? What am I missing?